src/three-domain-secure/component.jsx

```diff
@@ -115,9 +115,18 @@ export function getThreeDomainSecureComponent(): TDSComponent {
         onSuccess: {
           type: "function",
           alias: "onContingencyResult",
-          decorate: ({ value, onError }) => {
+          decorate: ({ props, value, onError }) => {
             return (err, result) => {
-              if (err || (result && !result.success)) {
+              const isCardFieldFlow = props.userType === "UNBRANDED_GUEST";
```

Comment:
Looks great!!! 

Would this handle if props are missing in BCDC usecases?
`props?.userType` maybe?